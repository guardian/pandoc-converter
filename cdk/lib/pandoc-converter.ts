import { GuEc2App } from '@guardian/cdk';
import { AccessScope } from '@guardian/cdk/lib/constants';
import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
import { GuStack } from '@guardian/cdk/lib/constructs/core';
import { GuCname } from '@guardian/cdk/lib/constructs/dns';
import type { App } from 'aws-cdk-lib';
import { Duration } from 'aws-cdk-lib';
import { InstanceClass, InstanceSize, InstanceType } from 'aws-cdk-lib/aws-ec2';

export class PandocConverter extends GuStack {
	converter: GuEc2App;
	cname: GuCname;

	constructor(scope: App, id: string, props: GuStackProps) {
		super(scope, id, props);

		const domainName = 'pandoc-converter.gutools.co.uk';

		this.converter = new GuEc2App(this, {
			access: {
				scope: AccessScope.PUBLIC,
			},
			app: 'pandoc-converter',
			applicationPort: 9482,
			certificateProps: {
				domainName,
			},
			imageRecipe: 'pandoc-converter-ubuntu-jammy-x86',
			instanceType: InstanceType.of(InstanceClass.T3, InstanceSize.NANO),
			monitoringConfiguration: { noMonitoring: true },
			scaling: {
				minimumInstances: 1,
			},
			userData: {
				distributable: {
					executionStatement:
						'converterPath=$(sudo nix-store --import < /pandoc-converter/pandoc-converter.closure | grep PandocConverter); $converterPath/bin/PandocConverter',
					fileName: 'pandoc-converter.closure',
				},
			},
		});

		this.cname = new GuCname(this, 'PandocConverterDNS', {
			app: this.converter,
			ttl: Duration.hours(1),
			domainName,
			resourceRecord: this.converter.loadBalancer.loadBalancerDnsName,
		});
	}
}
