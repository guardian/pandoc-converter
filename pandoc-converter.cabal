cabal-version:      3.4
name:               PandocConverter
version:            1.0.0
author:             Emily Bourke
maintainer:         emily@emdash.ie
category:           Application
build-type:         Simple

common warnings
    ghc-options: -Wall

executable PandocConverter
    import:           warnings
    main-is:          Main.hs
    other-modules:
      Composer
    -- other-extensions:
    build-depends:
        aeson,
        base,
        bytestring,
        -- cmdargs,
        containers,
        generic-lens,
        lens,
        megaparsec,
        pandoc,
        servant-server,
        servant-options,
        text,
        time,
        transformers,
        wai,
        warp,

    hs-source-dirs:   src
    default-language: GHC2021

-- Local Variables:
-- compile-command: "cabal2nix . > pandoc-converter.nix"
-- End:
