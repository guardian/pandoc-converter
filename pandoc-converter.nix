{ mkDerivation, aeson, base, bytestring, containers, generic-lens
, lens, lib, megaparsec, pandoc, servant-options, servant-server
, text, time, transformers, wai, wai-cors, warp
}:
mkDerivation {
  pname = "PandocConverter";
  version = "1.0.0";
  src = ./.;
  isLibrary = false;
  isExecutable = true;
  executableHaskellDepends = [
    aeson base bytestring containers generic-lens lens megaparsec
    pandoc servant-options servant-server text time transformers wai
    wai-cors warp
  ];
  license = "unknown";
  mainProgram = "PandocConverter";
}
