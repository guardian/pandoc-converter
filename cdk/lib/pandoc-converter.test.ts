import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { PandocConverter } from "./pandoc-converter";

describe("The PandocConverter stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new PandocConverter(app, "PandocConverter", { stack: "content-api", stage: "TEST" });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
