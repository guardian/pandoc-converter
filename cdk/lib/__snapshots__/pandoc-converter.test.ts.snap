// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PandocConverter stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuVpcParameter",
      "GuSubnetListParameter",
      "GuSubnetListParameter",
      "GuEc2App",
      "GuDistributionBucketParameter",
      "GuCertificate",
      "GuInstanceRole",
      "GuSsmSshPolicy",
      "GuDescribeEC2Policy",
      "GuLoggingStreamNameParameter",
      "GuLogShippingPolicy",
      "GuGetDistributablePolicy",
      "GuParameterStoreReadPolicy",
      "GuAmiParameter",
      "GuHttpsEgressSecurityGroup",
      "GuWazuhAccess",
      "GuAutoScalingGroup",
      "GuApplicationLoadBalancer",
      "GuApplicationTargetGroup",
      "GuHttpsApplicationListener",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "LoadBalancerPandocconverterDnsName": {
      "Description": "DNS entry for LoadBalancerPandocconverter",
      "Value": {
        "Fn::GetAtt": [
          "LoadBalancerPandocconverterA9DF7F8D",
          "DNSName",
        ],
      },
    },
  },
  "Parameters": {
    "AMIPandocconverter": {
      "Description": "Amazon Machine Image ID for the app pandoc-converter. Use this in conjunction with AMIgo to keep AMIs up to date.",
      "Type": "AWS::EC2::Image::Id",
    },
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "LoggingStreamName": {
      "Default": "/account/services/logging.stream.name",
      "Description": "SSM parameter containing the Name (not ARN) on the kinesis stream",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "VpcId": {
      "Default": "/account/vpc/primary/id",
      "Description": "Virtual Private Cloud to run EC2 instances within. Should NOT be the account default VPC.",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::VPC::Id>",
    },
    "pandocconverterPrivateSubnets": {
      "Default": "/account/vpc/primary/subnets/private",
      "Description": "A list of private subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
    "pandocconverterPublicSubnets": {
      "Default": "/account/vpc/primary/subnets/public",
      "Description": "A list of public subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
  },
  "Resources": {
    "AutoScalingGroupPandocconverterASGBCE81415": {
      "Properties": {
        "HealthCheckGracePeriod": 120,
        "HealthCheckType": "ELB",
        "LaunchTemplate": {
          "LaunchTemplateId": {
            "Ref": "contentapiTESTpandocconverterBAC92101",
          },
          "Version": {
            "Fn::GetAtt": [
              "contentapiTESTpandocconverterBAC92101",
              "LatestVersionNumber",
            ],
          },
        },
        "MaxSize": "2",
        "MetricsCollection": [
          {
            "Granularity": "1Minute",
          },
        ],
        "MinSize": "1",
        "Tags": [
          {
            "Key": "App",
            "PropagateAtLaunch": true,
            "Value": "pandoc-converter",
          },
          {
            "Key": "gu:cdk:version",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "PropagateAtLaunch": true,
            "Value": "guardian/pandoc-converter",
          },
          {
            "Key": "LogKinesisStreamName",
            "PropagateAtLaunch": true,
            "Value": {
              "Ref": "LoggingStreamName",
            },
          },
          {
            "Key": "Stack",
            "PropagateAtLaunch": true,
            "Value": "content-api",
          },
          {
            "Key": "Stage",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
        ],
        "TargetGroupARNs": [
          {
            "Ref": "TargetGroupPandocconverter32288488",
          },
        ],
        "VPCZoneIdentifier": {
          "Ref": "pandocconverterPrivateSubnets",
        },
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
    },
    "CertificatePandocconverterEF39FF61": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "DomainName": "pandoc-converter.gutools.co.uk",
        "Tags": [
          {
            "Key": "App",
            "Value": "pandoc-converter",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/pandoc-converter",
          },
          {
            "Key": "Name",
            "Value": "PandocConverter/CertificatePandocconverter",
          },
          {
            "Key": "Stack",
            "Value": "content-api",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
      "UpdateReplacePolicy": "Retain",
    },
    "DescribeEC2PolicyFF5F9295": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "autoscaling:DescribeAutoScalingInstances",
                "autoscaling:DescribeAutoScalingGroups",
                "ec2:DescribeTags",
                "ec2:DescribeInstances",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "describe-ec2-policy",
        "Roles": [
          {
            "Ref": "InstanceRolePandocconverter87FA087D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GetDistributablePolicyPandocconverter5D825BE1": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:s3:::",
                    {
                      "Ref": "DistributionBucketName",
                    },
                    "/content-api/TEST/pandoc-converter/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GetDistributablePolicyPandocconverter5D825BE1",
        "Roles": [
          {
            "Ref": "InstanceRolePandocconverter87FA087D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GuHttpsEgressSecurityGroupPandocconverter0FC6F0C1": {
      "Properties": {
        "GroupDescription": "Allow all outbound HTTPS traffic",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound HTTPS traffic",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "pandoc-converter",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/pandoc-converter",
          },
          {
            "Key": "Stack",
            "Value": "content-api",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "GuHttpsEgressSecurityGroupPandocconverterfromPandocConverterLoadBalancerPandocconverterSecurityGroupB57C3E5E94826AFC8E94": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 9482,
        "GroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupPandocconverter0FC6F0C1",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "LoadBalancerPandocconverterSecurityGroup5752B8E3",
            "GroupId",
          ],
        },
        "ToPort": 9482,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "GuLogShippingPolicy981BFE5A": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "kinesis:Describe*",
                "kinesis:Put*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:kinesis:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":stream/",
                    {
                      "Ref": "LoggingStreamName",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GuLogShippingPolicy981BFE5A",
        "Roles": [
          {
            "Ref": "InstanceRolePandocconverter87FA087D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "InstanceRolePandocconverter87FA087D": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Path": "/",
        "Tags": [
          {
            "Key": "App",
            "Value": "pandoc-converter",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/pandoc-converter",
          },
          {
            "Key": "Stack",
            "Value": "content-api",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ListenerPandocconverterF61A52D7": {
      "Properties": {
        "Certificates": [
          {
            "CertificateArn": {
              "Ref": "CertificatePandocconverterEF39FF61",
            },
          },
        ],
        "DefaultActions": [
          {
            "TargetGroupArn": {
              "Ref": "TargetGroupPandocconverter32288488",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": {
          "Ref": "LoadBalancerPandocconverterA9DF7F8D",
        },
        "Port": 443,
        "Protocol": "HTTPS",
        "SslPolicy": "ELBSecurityPolicy-TLS13-1-2-2021-06",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "LoadBalancerPandocconverterA9DF7F8D": {
      "Properties": {
        "LoadBalancerAttributes": [
          {
            "Key": "deletion_protection.enabled",
            "Value": "true",
          },
          {
            "Key": "routing.http.x_amzn_tls_version_and_cipher_suite.enabled",
            "Value": "true",
          },
          {
            "Key": "routing.http.drop_invalid_header_fields.enabled",
            "Value": "true",
          },
        ],
        "Scheme": "internet-facing",
        "SecurityGroups": [
          {
            "Fn::GetAtt": [
              "LoadBalancerPandocconverterSecurityGroup5752B8E3",
              "GroupId",
            ],
          },
        ],
        "Subnets": {
          "Ref": "pandocconverterPublicSubnets",
        },
        "Tags": [
          {
            "Key": "App",
            "Value": "pandoc-converter",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/pandoc-converter",
          },
          {
            "Key": "Stack",
            "Value": "content-api",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "LoadBalancerPandocconverterSecurityGroup5752B8E3": {
      "Properties": {
        "GroupDescription": "Automatically created Security Group for ELB PandocConverterLoadBalancerPandocconverter006202B7",
        "SecurityGroupIngress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow from anyone on port 443",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "pandoc-converter",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/pandoc-converter",
          },
          {
            "Key": "Stack",
            "Value": "content-api",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "LoadBalancerPandocconverterSecurityGrouptoPandocConverterGuHttpsEgressSecurityGroupPandocconverterA5359A009482147348C5": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupPandocconverter0FC6F0C1",
            "GroupId",
          ],
        },
        "FromPort": 9482,
        "GroupId": {
          "Fn::GetAtt": [
            "LoadBalancerPandocconverterSecurityGroup5752B8E3",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 9482,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "LoadBalancerPandocconverterSecurityGrouptoPandocConverterWazuhSecurityGroupEAF958DF948217E9816C": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "FromPort": 9482,
        "GroupId": {
          "Fn::GetAtt": [
            "LoadBalancerPandocconverterSecurityGroup5752B8E3",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 9482,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "ParameterStoreReadPandocconverterA144CDA6": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/content-api/pandoc-converter",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/content-api/pandoc-converter/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "parameter-store-read-policy",
        "Roles": [
          {
            "Ref": "InstanceRolePandocconverter87FA087D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SsmSshPolicy4CFC977E": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "ec2messages:AcknowledgeMessage",
                "ec2messages:DeleteMessage",
                "ec2messages:FailMessage",
                "ec2messages:GetEndpoint",
                "ec2messages:GetMessages",
                "ec2messages:SendReply",
                "ssm:UpdateInstanceInformation",
                "ssm:ListInstanceAssociations",
                "ssm:DescribeInstanceProperties",
                "ssm:DescribeDocumentParameters",
                "ssmmessages:CreateControlChannel",
                "ssmmessages:CreateDataChannel",
                "ssmmessages:OpenControlChannel",
                "ssmmessages:OpenDataChannel",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ssm-ssh-policy",
        "Roles": [
          {
            "Ref": "InstanceRolePandocconverter87FA087D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TargetGroupPandocconverter32288488": {
      "Properties": {
        "HealthCheckIntervalSeconds": 10,
        "HealthCheckPath": "/healthcheck",
        "HealthCheckProtocol": "HTTP",
        "HealthCheckTimeoutSeconds": 5,
        "HealthyThresholdCount": 5,
        "Port": 9482,
        "Protocol": "HTTP",
        "Tags": [
          {
            "Key": "App",
            "Value": "pandoc-converter",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/pandoc-converter",
          },
          {
            "Key": "Stack",
            "Value": "content-api",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "TargetGroupAttributes": [
          {
            "Key": "deregistration_delay.timeout_seconds",
            "Value": "30",
          },
          {
            "Key": "stickiness.enabled",
            "Value": "false",
          },
        ],
        "TargetType": "instance",
        "UnhealthyThresholdCount": 2,
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "WazuhSecurityGroup": {
      "Properties": {
        "GroupDescription": "Allow outbound traffic from wazuh agent to manager",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh event logging",
            "FromPort": 1514,
            "IpProtocol": "tcp",
            "ToPort": 1514,
          },
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh agent registration",
            "FromPort": 1515,
            "IpProtocol": "tcp",
            "ToPort": 1515,
          },
        ],
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/pandoc-converter",
          },
          {
            "Key": "Stack",
            "Value": "content-api",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "WazuhSecurityGroupfromPandocConverterLoadBalancerPandocconverterSecurityGroupB57C3E5E9482E1A59F98": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 9482,
        "GroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "LoadBalancerPandocconverterSecurityGroup5752B8E3",
            "GroupId",
          ],
        },
        "ToPort": 9482,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "contentapiTESTpandocconverterBAC92101": {
      "DependsOn": [
        "InstanceRolePandocconverter87FA087D",
      ],
      "Properties": {
        "LaunchTemplateData": {
          "IamInstanceProfile": {
            "Arn": {
              "Fn::GetAtt": [
                "contentapiTESTpandocconverterProfileFB78134A",
                "Arn",
              ],
            },
          },
          "ImageId": {
            "Ref": "AMIPandocconverter",
          },
          "InstanceType": "t3.nano",
          "MetadataOptions": {
            "HttpTokens": "required",
            "InstanceMetadataTags": "enabled",
          },
          "SecurityGroupIds": [
            {
              "Fn::GetAtt": [
                "GuHttpsEgressSecurityGroupPandocconverter0FC6F0C1",
                "GroupId",
              ],
            },
            {
              "Fn::GetAtt": [
                "WazuhSecurityGroup",
                "GroupId",
              ],
            },
          ],
          "TagSpecifications": [
            {
              "ResourceType": "instance",
              "Tags": [
                {
                  "Key": "App",
                  "Value": "pandoc-converter",
                },
                {
                  "Key": "gu:cdk:version",
                  "Value": "TEST",
                },
                {
                  "Key": "gu:repo",
                  "Value": "guardian/pandoc-converter",
                },
                {
                  "Key": "Name",
                  "Value": "PandocConverter/content-api-TEST-pandoc-converter",
                },
                {
                  "Key": "Stack",
                  "Value": "content-api",
                },
                {
                  "Key": "Stage",
                  "Value": "TEST",
                },
              ],
            },
            {
              "ResourceType": "volume",
              "Tags": [
                {
                  "Key": "App",
                  "Value": "pandoc-converter",
                },
                {
                  "Key": "gu:cdk:version",
                  "Value": "TEST",
                },
                {
                  "Key": "gu:repo",
                  "Value": "guardian/pandoc-converter",
                },
                {
                  "Key": "Name",
                  "Value": "PandocConverter/content-api-TEST-pandoc-converter",
                },
                {
                  "Key": "Stack",
                  "Value": "content-api",
                },
                {
                  "Key": "Stage",
                  "Value": "TEST",
                },
              ],
            },
          ],
          "UserData": {
            "Fn::Base64": {
              "Fn::Join": [
                "",
                [
                  "#!/bin/bash
mkdir -p $(dirname '/pandoc-converter/pandoc-converter.closure')
aws s3 cp 's3://",
                  {
                    "Ref": "DistributionBucketName",
                  },
                  "/content-api/TEST/pandoc-converter/pandoc-converter.closure' '/pandoc-converter/pandoc-converter.closure'
converterPath=$(sudo nix-store --import < /pandoc-converter/pandoc-converter.closure | grep PandocConverter); $converterPath/bin/PandocConverter",
                ],
              ],
            },
          },
        },
        "TagSpecifications": [
          {
            "ResourceType": "launch-template",
            "Tags": [
              {
                "Key": "App",
                "Value": "pandoc-converter",
              },
              {
                "Key": "gu:cdk:version",
                "Value": "TEST",
              },
              {
                "Key": "gu:repo",
                "Value": "guardian/pandoc-converter",
              },
              {
                "Key": "Name",
                "Value": "PandocConverter/content-api-TEST-pandoc-converter",
              },
              {
                "Key": "Stack",
                "Value": "content-api",
              },
              {
                "Key": "Stage",
                "Value": "TEST",
              },
            ],
          },
        ],
      },
      "Type": "AWS::EC2::LaunchTemplate",
    },
    "contentapiTESTpandocconverterProfileFB78134A": {
      "Properties": {
        "Roles": [
          {
            "Ref": "InstanceRolePandocconverter87FA087D",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
  },
}
`;
